src/orquestra/sdk/_base/_dsl.py

```diff
@@ -480,6 +486,40 @@ def __call__(self, *args: _P.args, **kwargs: _P.kwargs) -> _R:
             ),
         )
 
+    def resolve_task_source_data(
+        self, wf_default_source_import: Optional[Import] = None
+    ):
+        # if user set source import explicitly, do nothing
+        if self._use_default_source_import:
+            if wf_default_source_import:
+                self.source_import = wf_default_source_import
+            # Set the default Import based on if the session is interactive
+            elif _is_interactive():
+                self.source_import = InlineImport()
+            else:
+                self.source_import = LocalImport(module=self.__sdk_task_body.__module__)
+        self._resolve_fn_ref()
+
+    def _resolve_fn_ref(self):
+        # resolve fn_ref is based on task source import. If user doesn't pass it,
+        # resolve_source_import should set it
+        assert self.source_import is not None
+        self.fn_ref = (
+            InlineFunctionRef(self.__sdk_task_body.__name__, self.__sdk_task_body)
+            if isinstance(self.source_import, InlineImport)
+            else get_fn_ref(self.__sdk_task_body)
+        )
+
+    def resolve_task_dependencies(
+        self, wf_default_dependency_imports: Optional[Tuple[Import, ...]] = None
+    ):
+        # if user set imports explicitly, do nothing
+        if not self._use_default_dependency_imports:
+            return
+
+        if wf_default_dependency_imports:
+            self.dependency_imports = wf_default_dependency_imports
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏Should this be private?